frontend/components/idioms.jsx

```diff
@@ -145,216 +126,309 @@ export function UntraceableState() {
+      ))}
+    </ul>
+  );
 }
 
 export function SubstandardDataStructure() {
-  const [error, setError] = useState("")
+  const Errors = {
+    ERROR_A: 'Error A',
+    ERROR_B: 'Error B',
+    NO_ERROR: '',
+  };
+  const [error, setError] = useState(Errors.NO_ERROR); // set data structure to Enums
 
-  return(
+  return (
     <div>
-      <button onClick={() => setError('Error A')}>Throw Error A</button>
-      <button onClick={() => setError('Error B')}>Throw Error B</button>
-      <button onClick={() => setError('')}>Clear Errors</button>
-      <div>
-        {error}
-      </div>
+      <button onClick={() => setError(Errors.ERROR_A)}>Throw Error A</button>
+      <button onClick={() => setError(Errors.ERROR_B)}>Throw Error B</button>
+      <button onClick={() => setError(Errors.NO_ERROR)}>Clear Errors</button>
+      <div>{error}</div>
     </div>
-  )
+  );
 }
 
 export function DangerousIdentifier() {
-  const [people, setPeople] = useState([])
+  const [people, setPeople] = useState([]);
 
   const handleSubmit = (e) => {
-    e.preventDefault()
+    e.preventDefault();
     const person = new FormData(e.target);
-    setPeople(ppl => [...ppl, ...person])
-  }
+    setPeople((ppl) => [...ppl, ...person]);
+  };
 
-  return(
+  return (
     <div>
       <form onSubmit={handleSubmit}>
-        <input type="text" />
+        <input type='text' />
         <button>Add Person</button>
       </form>
       <ul>
-        {people.map(person => <span key={person.name}>{person.name}</span>)}
+        {people.map((person, index) => (
+          <span key={person.name + index}>{person.name}</span> // people could have the same name
+        ))}
       </ul>
     </div>
-  )
+  );
 }
 
-async function fetchLeader() { return { name: 'Messi' }}
-async function fetchDetails(leader) { return { ...leader, country: 'Argentina' }}
+async function fetchLeader() {
+  return { name: 'Messi' };
+}
+async function fetchDetails(leader) {
+  return { ...leader, country: 'Argentina' };
+}
 
 // Hint: this only requires a single line change!
 export function UnnecessaryEffectTriggering() {
-  const [leader, setLeader] = useState({})
+  const [leader, setLeader] = useState({});
 
   useEffect(() => {
     const interval = setInterval(async () => {
-      const leader = await fetchLeader()
-      setLeader(leader)
-    }, 1000)
-    return () => clearInterval(interval)
-  }, [])
+      const leader = await fetchLeader();
+      setLeader(leader);
+    }, 1000);
+    return () => clearInterval(interval);
+  }, []);
 
   useEffect(() => {
+    if (!leader) return; //leader will be empty at the begging
     async function enhanceRecord() {
       const enriched = await fetchDetails(leader);
       setLeader(enriched);
     }
     enhanceRecord();
-  }, [leader]); 
-
+  }, [leader]);
 
-  return(
+  return (
     <div>
       <div>Leader:{leader.name}</div>
       {leader.country && <div>{`From: ${leader.country}`}</div>}
     </div>
-  )
+  );
```

Comment:
Very wrong!

Reply:
Would it be `if(!empty(leader)) enhanceRecord();`?